airflow/providers/amazon/aws/hooks/ssm.py

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class SsmHook(AwsBaseHook):
+    """
+    Interact with Amazon Systems Manager (SSM) using the boto3 library.
+    All API calls available through the Boto API are also available here.
+    See: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ssm.html#client
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["client_type"] = "ssm"
+        super().__init__(*args, **kwargs)
+
+        self.ParameterNotFoundException = self.conn.exceptions.ParameterNotFound
```

Comment:
I don't think we should create this in Hook constructor because it immediately call Airflow connection

Reply:
How do you feel about moving it to a @property:

```
class SsmHook(AwsBaseHook):
    def __init__(self, *args, **kwargs) -> None:
        kwargs["client_type"] = "ssm"
        super().__init__(*args, **kwargs)
    
    @property
    def ParameterNotFoundException(self):
        return self.conn.exceptions.ParameterNotFound
```

Reply:
Yeah, I did it this way specifically to avoid unwrapping the ClientError but that's an option too.  

I'll do it as a property, but I'd like to keep it CamelCase since exception names are... well... exceptions to that rule and are (always?) CamelCase in python.  Would you object to leaving it that way?

Reply:
BTW, any concern why we can just use

```python
try:
    return self.conn.get_parameter(Name=parameter)["Parameter"]["Value"]
except self.conn.exceptions.ParameterNotFound as ex:
    ...
```

Reply:
Done.  Required a small tweak in the unit tests to match.